1) For **two independent** strategy-return samples, the Mann–Whitney U test can test whether their return distributions differ in location/rank.

In [1]:
import numpy as np
from scipy.stats import mannwhitneyu


# ============================================================
# 1. Daily returns of two strategies
# ============================================================

strategy_a = np.array([
    0.012, 0.008, 0.015, -0.003, 0.010,
    0.007, -0.002, 0.013, 0.009, 0.011,
    0.006, 0.010, 0.004, 0.012, 0.008
])

strategy_b = np.array([
    0.005, 0.010, 0.004, 0.002, 0.006,
    0.003, 0.001, 0.005, 0.007, 0.004,
    0.003, 0.006, 0.005, 0.004, 0.006
])


# ============================================================
# 2. Hypotheses
# ============================================================

# H0: The two strategies have the same distribution/location.
#
# H1: The two strategies have different distributions/location.


alpha = 0.05


# ============================================================
# 3. Descriptive statistics
# ============================================================

mean_a = np.mean(strategy_a)
mean_b = np.mean(strategy_b)

median_a = np.median(strategy_a)
median_b = np.median(strategy_b)

print("========== STRATEGY RETURNS ==========")

print(f"Strategy A mean   : {mean_a:.4%}")
print(f"Strategy B mean   : {mean_b:.4%}")

print(f"Strategy A median : {median_a:.4%}")
print(f"Strategy B median : {median_b:.4%}")


# ============================================================
# 4. Mann-Whitney U Test
# ============================================================

u_statistic, p_value = mannwhitneyu(
    strategy_a,
    strategy_b,
    alternative="two-sided"
)


print("\n========== MANN-WHITNEY U TEST ==========")

print(f"U statistic : {u_statistic:.4f}")
print(f"p-value     : {p_value:.4f}")
print(f"alpha       : {alpha}")


# ============================================================
# 5. Statistical Decision
# ============================================================

if p_value <= alpha:

    print("\nDecision: REJECT H0")

    print(
        "There is statistically significant evidence that "
        "the return distributions of Strategy A and Strategy B "
        "differ."
    )

else:

    print("\nDecision: FAIL TO REJECT H0")

    print(
        "There is insufficient statistical evidence to conclude "
        "that the return distributions of Strategy A and Strategy B "
        "differ."
    )

========== STRATEGY RETURNS ==========
Strategy A mean   : 0.8000%
Strategy B mean   : 0.4733%
Strategy A median : 0.9000%
Strategy B median : 0.5000%

========== MANN-WHITNEY U TEST ==========
U statistic : 176.5000
p-value     : 0.0082
alpha       : 0.05

Decision: REJECT H0
There is statistically significant evidence that the return distributions of Strategy A and Strategy B differ.


If you specifically want to test whether A is better than B

Use a **one-sided test**:

In [ ]:
u_statistic, p_value = mannwhitneyu(
    strategy_a,
    strategy_b,
    alternative="greater"
)

Here the hypotheses are:

H0: Strategy A is not systematically higher than B
H1​: Strategy A tends to have higher returns than B
Then:

In [2]:
if p_value <= 0.05:
    print("Reject H0")
    print("Evidence suggests Strategy A has higher returns.")
else:
    print("Fail to reject H0")
    print("Insufficient evidence that Strategy A has higher returns.")

Reject H0
Evidence suggests Strategy A has higher returns.


Example interpretation

Suppose your output is:

Strategy A mean   : 0.8533%
Strategy B mean   : 0.4800%

Strategy A median : 0.9000%
Strategy B median : 0.5000%

U statistic : 198.0000
p-value     : 0.0120
alpha       : 0.05

Decision: REJECT H0

Your conclusion would be:

**The Mann–Whitney U test produced a p-value of 0.012, which is less than the 5% significance level. Therefore, we reject the null hypothesis and conclude that there is statistically significant evidence that the return distributions of Strategy A and Strategy B differ.**